pkgs/by-name/dc/dc3dd/package.nix

```diff
@@ -0,0 +1,38 @@
+{ lib
+, stdenv
+, fetchzip
+, perl538Packages
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "dc3dd";
+  version = "7.3.1";
+
+  src = fetchzip {
+      url = "http://downloads.sf.net/sourceforge/dc3dd/dc3dd-${finalAttrs.version}.zip";
+      hash = "sha256-SYDoqGlsROHX1a0jJX11F+yp6CeFK+tZbYOOnScC6Ig=";
+  };
+
+  postPatch = ''
+    chmod +x configure
+  '';
+
+  buildInputs = [ perl538Packages.LocaleGettext ];
+
+  makeFlags = [
+    "PREFIX=$out"
+  ];
+
+  configurePlatforms = [ "build" ];
```

Comment:
Why?

Reply:
I was studying the Nix manual for **configurePlatforms** that provides **build**. According to the build information of the tool in [BlackArch](https://github.com/BlackArch/blackarch/blob/master/packages/dc3dd/PKGBUILD), I see that for non-i686 and non-x86_64, it invokes `./configure --build=arm`, so I'm thinking the correct usage for configure Platforms is to use it as `configurePlatforms = [ "build" ];`.

About `"PREFIX=$out"` in `makeFlags`, if it is by default set as `$out` when not set, I can delete it.

Reply:
Let me know if I am understanding correctly:

The configuration script accepts a command-line option called `--build`.
In ARM architectures, it should be set to `--build=arm`.

If the above is correct, then you should use the code below:

```nix
configureFlags = lib.optionals (stdenv.hostPlatform.isAarch32 || stdenv.hostPlatform.isAarch64) [ "--build=arm" ];
```

Reply:
That's correct

Reply:
Then you can remove the line above.

Reply:
Sorry for my mistakes. Thank you for the suggestions.